src/serac/physics/materials/thermal_expansion_material.hpp

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2019-2021, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/**
+ * @file hyperelastic_material.hpp
+ *
+ * @brief The hyperelastic material models for the solid module
+ */
+
+#pragma once
+
+#include "mfem.hpp"
+#include "serac/physics/utilities/finite_element_state.hpp"
+
+namespace serac {
+
+/**
+ * @brief Abstract interface class for a generic hyperelastic material
+ *
+ */
+class ThermalExpansionMaterial {
+public:
+  /**
+   * @brief Construct a new Hyperelastic Material object
+   *
+   */
+  ThermalExpansionMaterial() : parent_to_reference_transformation_(nullptr) {}
+
+  /**
+   * @brief Destroy the Hyperelastic Material object
+   *
+   */
+  virtual ~ThermalExpansionMaterial() = default;
+
+  /**
+   * @brief Set the reference-to-target transformation. This is required to use coefficient parameters.
+   *
+   * @param[in] Ttr The reference-to-target (stress-free) transformation
+   */
+  void setTransformation(mfem::ElementTransformation& Ttr) { parent_to_reference_transformation_ = &Ttr; }
+
+  /**
+   * @brief Evaluate the strain energy density function, W = W(F).
+   *
+   * @param[in] du_dX the displacement gradient
+   * @return double Strain energy density
+   */
+  virtual void evalThermalDeformationGradient(mfem::DenseMatrix& dx_dX) = 0;
+
+protected:
+  /**
+   * @brief Non-owning pointer to the reference element to stree-free configuration (target) transformation
+   *
+   */
+  mfem::ElementTransformation* parent_to_reference_transformation_;
+};
+
+/**
+ * @brief Neo-Hookean hyperelastic model with a strain energy density function given
+ *   by the formula: \f$(\mu/2)(\bar{I}_1 - dim) + (K/2)(det(F)/g - 1)^2\f$ where
+ *   F is the deformation gradient and \f$\bar{I}_1 = (det(F))^{-2/dim} Tr(F
+ *   F^t)\f$. The parameters \f$\mu\f$ and \f$\lambda\f$ are the Lame parameters.
```

Comment:
Am I correct in understanding that the purpose of the `ThermalExpansionMaterial` is to  provide an appropriate `evalThermalDeformationGradient` method that can be used to augment the (total) thermal deformation gradient within the `DisplacementHyperelasticIntegrator`? If so, I think this Doxygen needs to be updated since it looks like the `ThermalExpansionMaterial` does not provide a complete SEDF on its own.

Reply:
Oh yes - the doxygen is 100% wrong and simply an artifact of my cutting and pasting. I'm doing a pass now to correct it.